src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -41,26 +42,38 @@ public IReadOnlyCollection<ToolManifestPackage> Find(FilePath? filePath = null)
             var result = new List<ToolManifestPackage>();
             foreach (FilePath possibleManifest in allPossibleManifests)
             {
-                if (_fileSystem.File.Exists(possibleManifest.Value))
+                if (!_fileSystem.File.Exists(possibleManifest.Value))
                 {
-                    findAnyManifest = true;
-                    SerializableLocalToolsManifest deserializedManifest =
-                        DeserializeLocalToolsManifest(possibleManifest);
+                    continue;
+                }
 
-                    foreach (ToolManifestPackage p in GetToolManifestPackageFromOneManifestFile(deserializedManifest, possibleManifest))
-                    {
-                        if (!result.Any(addedToolManifestPackages =>
-                            addedToolManifestPackages.PackageId.Equals(p.PackageId)))
-                        {
-                            result.Add(p);
-                        }
-                    }
+                findAnyManifest = true;
+                SerializableLocalToolsManifest deserializedManifest =
+                    DeserializeLocalToolsManifest(possibleManifest);
```

Comment:
Is this expected to propagate a `JsonReaderException` to the caller?  I see a catch around the call to `DeserializeLocalToolsManifest` in `TryFind` below.

On a related note: perhaps this method should now be called `FindAll` since `TryFind` returns a singular `ToolManifestPackage` while this returns a collection?

Reply:
Yes, and `GivenInvalidFieldsManifestFileWhenFindByCommandNameItReturnFalse` is covering this case.

Command resolver is kind of like chain of responsibility. And if the previous resolver returns null it will try to use the next resolver. So it should not throw for failure of resolving command. And if the user try to run a, say project tools, but there is an invalid manifest file on the disk. If we throw here, the user can never get to use the project tool.

In this PR  I do throw on nuget cache being cleaned. I hope to discuss this on 3pm's meeting since I am not sure we should throw or auto restore or being silent

Reply:
Per the discussion on Monday, i changed it to throw error